packages/core/src/utils.ts

```diff
@@ -153,6 +153,11 @@ export function readFileAsStringSync(filePath: string) {
   return fs.readFileSync(filePath, 'utf-8');
 }
 
+export function readAndParseFileSync<T>(filePath: string) {
```

Comment:
Is it only for yaml and json files? If so, it worth renaming this functions to be explicit.